### PR TITLE
Load T1w-to-standard transform to same space as volumetric BOLD scan

### LINKED ISSUE #925: CIFTI run with Nibabies data collected MNIInfant for BOLD and MNI152NLin6Asym for T1w
## Summary

Per #759, XCP-D should collect the MNI152NLin6Asym NIFTI for the executive summary when the input-type is Nibabies and the `--cifti` flag is used. However, based on a bug report by @madisoth and @lundq163, it looks like the executive summary plotted a mean BOLD image in MNIInfant space and a T1w in MNI152NLin6Asym.

## Additional details

- xcp_d version: 0.4.0
- Docker version:
- Singularity version:

### What were you trying to do?

```
singularity run --cleanenv \
    xcp_d_unstable_07032023a.sif /data /out participant \
    --input-type nibabies \
    --cifti \
    --despike \
    --resource-monitor \
    --dcan-qc \
    -w /work \
    --omp-nthreads 3 \
    --warp-surfaces-native2std \
    -f 0.3 \
    -m \
    -vv \
    --participant-label ??????
```

### PR DESCRIPTION
**Pull request by tsalo**
Closes #925.

## Changes proposed in this pull request

- ~~Add an `inverted` flag to `get_std2bold_xfms` to get transforms _to_ MNI152NLin6Asym instead of from it.~~
- ~~Add steps in `init_execsummary_functional_plots_wf` to warp the mean BOLD image and the BOLD reference image to MNI152NLin6Asym (which is the space the T1w/T2w should be in) before plotting them.~~
- Make MNI152NLin6Asym the preferred space for nibabies data.
- Use the template from the anat-to-template transform to select the appropriate NIfTI preprocessed BOLD and boldref for cifti runs with the executive summary enabled.
    - This means we need to ensure that there are BOLD files available in the target space, which we now do.

### CODE REVIEW

**Comment by tsalo** [comment]
@madisoth and @lundq163, could on of you pull the branch associated with this PR and see if it fixes the issue?

**Comment by tsalo** [comment]
Per @lundq163, the problem is still present with the current changes.

**Comment by lundq163** [comment]
hi @tsalo, i just retested it again and am seeing the same issue as before. here are the paths for the output and work dirs on MSI:
`/scratch.global/lundq163/mcribs_test/BCP_man_corr_asegs/derivatives_xcpd_0720`
`/scratch.global/lundq163/mcribs_test/BCP_man_corr_asegs/work_xcpd_0720`

and this was my run command:
`singularity run --cleanenv \
-B /home/faird/shared/code/external/utilities/freesurfer_license/license.txt:/opt/freesurfer/license.txt \
-B /scratch.global/lundq163/mcribs_test/BCP_man_corr_asegs/derivatives_goodvox_mcribs_0629_tm:/data:ro \
-B /scratch.global/lundq163/mcribs_test/BCP_man_corr_asegs/derivatives_xcpd_0720:/out \
-B /scratch.global/lundq163/mcribs_test/BCP_man_corr_asegs/work_xcpd_0720:/work \
-B /home/faird/shared/code/external/pipelines/xcp_d_test_binds/20230712_fix-collection/xcp_d/xcp_d:/usr/local/miniconda/lib/python3.8/site-packages/xcp_d \
/home/faird/shared/code/external/pipelines/xcp_d/xcp_d_unstable_07172023a.sif /data /out participant \
--input-type nibabies \
--cifti \
--despike \
--resource-monitor \
--dcan-qc \
-w /work \
--omp-nthreads 3 \
--warp-surfaces-native2std \
-f 0.3 \
-m \
-vv \
--participant-label 981854 \`

@madisoth and I took some time to look through the work dir, and he concluded that the command being used to warp the mean BOLD looks correct (`<...>/cifti_postprocess_<n>_wf/execsummary_functional_plots_wf/warp_mean_bold_to_mni/command.txt`) but the result looks the the opposite of what it should be: if transforming from MNIInfant to MNI adult space the brain should be getting bigger, not smaller. Thomas suggested maybe one or both of the transforms (MNIInfant -> MNI2009, MNI2009 -> MNI6) are misnamed and it's actually the inverse transform, or something along those lines? I will try to look into it more on my own time next week too. Lmk if you need me to do anymore testing though moving forward, thanks!

**Comment by tsalo** [comment]
I can't see anything wrong with the ApplyTransforms calls or the collection of the transforms, so I worry that maybe the MNIInfant-to-MNI152NLin2009cAsym transform is actually a copy of the MNI152NLin2009cAsym-to-MNIInfant transform.

File: xcp_d/utils/bids.py [hunk 1]
```diff
@@ -32,8 +32,8 @@
     "nibabies": {
         "cifti": ["fsLR"],
         "nifti": [
-            "MNIInfant",
             "MNI152NLin6Asym",
+            "MNIInfant",
             "MNI152NLin2009cAsym",
         ],
```
Comment (tsalo) [inline xcp_d/utils/bids.py:38]: Make MNI152NLin6Asym the preferred space for nibabies data.

**Comment by tsalo** [comment]
I believe the problem is just that the space selected for NIfTI BOLD and boldref files was independent from the space selected for the anat-to-template transform, which is what is applied to the T1w data.

ds001419_cifti has a T1w-to-MNI152NLin6Asym transform, but that's only because of AROMA. There are no MNI152NLin6Asym BOLD outputs, except `smoothAROMAnonaggr` files. Somehow I need to take the run data into account when selecting the target space across runs, which is real fun.

**Comment by tsalo** [comment]
I am now looping through allowed volumetric spaces for a given input type in order to find one with NIfTI BOLD files in that space when running with the `--cifti` flag. This _should_ ensure that the target space for the anatomical images is the same as the space the preprocessed BOLD NIfTIs are in, for the sake of the T1w-on-BOLD and related figures. 

I am concerned that the brainsprite overlaying the white matter and pial surfaces on top of the anatomicals won't work correctly if the anatomicals are in a space other than MNI152NLin6Asym. In the case of ds001419-cifti, the best available space is MNI152NLin2009cAsym, so I will compare the brainsprite from this PR's ds001419-cifti executive summary against the one from `main`, I guess.

**Comment by tsalo** [comment]
@madisoth @lundq163 would you mind checking the new version of this branch?

**Pull request merged**